### PR TITLE
[script] Enable debug server if ran from vscode terminal.

### PR DESCRIPTION
**Pull request by alloy**
Ok, spent a bit more time trying to come up with a good pattern to automatically attach vscode’s debugger in a re-usable way across projects. I’ll comment inline.

### CODE REVIEW

File: scripts/start.sh [hunk 1]
```diff
@@ -5,11 +5,23 @@ set -ex
...
+
+OPT=(--max_old_space_size=4096)
+
+if [ "$NODE_ENV" != "production" ]; then
   if [ ! -f "./.env" ]; then
     echo -e "\033[1;31m WARNING: Missing .env file, see CONTRIBUTING.md. \033[0m"
   fi
-  node -r dotenv/config --max_old_space_size=4096 ./src
+  OPT+=(-r dotenv/config)
+  if [ "$TERM_PROGRAM" == "vscode" ] && \
```
Comment (alloy) [inline scripts/start.sh]: Check if running in vscode’s terminal

File: scripts/start.sh [hunk 2]
```diff
@@ -5,11 +5,23 @@ set -ex
...
+     [ ! -z "$TERM_PROGRAM_VERSION" ] && \
```
Comment (alloy) [inline scripts/start.sh]: Check if the version var exists, just in case

File: scripts/start.sh [hunk 3]
```diff
@@ -5,11 +5,23 @@ set -ex
...
+     [ $(version $TERM_PROGRAM_VERSION) -ge $(version "1.22.2") ]
```
Comment (alloy) [inline scripts/start.sh]: Check if the vscode version is >= 1.22.2 which introduced the auto-attach feature

File: scripts/start.sh [hunk 4]
```diff
@@ -5,11 +5,23 @@ set -ex
...
+  then
+    OPT+=(--inspect-brk)
+  fi
 fi
+
+exec node "${OPT[@]}" ./src
```
Comment (alloy) [inline scripts/start.sh]: Spread options and `exec` rather than keeping shell process running

File: scripts/start.sh (hunk 1 above)
Comment (orta) [reply scripts/start.sh]: ohhh, interesting

**Comment by orta** [comment]
very cool 👍

**Pull request merged**